python-package/lightgbm/sklearn.py

```diff
@@ -479,7 +479,7 @@ def fit(self, X, y,
             eval_metric = [eval_metric] if isinstance(eval_metric, (string_type, type(None))) else eval_metric
             params['metric'] = set(original_metric + eval_metric)
 
-        if not isinstance(X, DataFrame):
+        if not isinstance(X, DataFrame) and not isinstance(X, DataTable):
```

Comment:
Just letting you know that it can be `if not isinstance(X, (DataFrame, DataTable)):` for short.